src/logging/checkpoint_manager.cpp

```diff
@@ -0,0 +1,450 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// checkpoint_manager.cpp
+//
+// Identification: src/backend/logging/checkpoint_manager.cpp
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "logging/checkpoint_manager.h"
+#include "logging/logging_util.h"
+#include "storage/database.h"
+#include "storage/data_table.h"
+#include "storage/tile_group.h"
+#include "storage/tile_group_header.h"
+#include "concurrency/transaction_manager.h"
+#include "concurrency/transaction_manager_factory.h"
+#include "concurrency/epoch_manager_factory.h"
+#include "type/ephemeral_pool.h"
+#include "common/timer.h"
+#include "storage/storage_manager.h"
+
+namespace peloton {
+namespace logging {
+
+  size_t CheckpointManager::DoRecovery() {
+
+    std::vector<std::vector<size_t>> database_structures;
+
+    size_t persist_epoch_id = RecoverPepoch(database_structures);
+    
+    // reload all the files with file name containing "persist epoch id"
+    RecoverCheckpoint(persist_epoch_id, database_structures);
+    return persist_epoch_id;
+  }
+
+  size_t CheckpointManager::RecoverPepoch(std::vector<std::vector<size_t>> &database_structures) {
```

Comment:
This is a really complex method but there is barely any comments about what is happening.